### PR TITLE
Create Bundle instead of building Index for FAR using operator-sdk

### PR DESCRIPTION
**Pull request by razo7**
Building an index is deprecated, and it supports moving to file-based catalog - [ECOPROJECT-1577](https://issues.redhat.com//browse/ECOPROJECT-1577)
Related links:

- [Building an Index: Deprecated](https://docs.ci.openshift.org/docs/how-tos/testing-operator-sdk-operators/#building-an-index-deprecated)
- [Step Registry Content for Operators](https://docs.ci.openshift.org/docs/how-tos/testing-operator-sdk-operators/#step-registry-content-for-operators)
- [Moving to File-Based Catalog](https://docs.ci.openshift.org/docs/how-tos/testing-operator-sdk-operators/#moving-to-file-based-catalog)

### CODE REVIEW

**Comment by razo7** [comment]
/pj-rehearse pull-ci-medik8s-fence-agents-remediation-main-4.12-openshift-e2e

**Comment by razo7** [comment]
/pj-rehearse pull-ci-medik8s-fence-agents-remediation-main-4.13-openshift-e2e

**Comment by razo7** [comment]
/pj-rehearse pull-ci-medik8s-fence-agents-remediation-main-4.13-openshift-e2e

**Comment by razo7** [comment]
/pj-rehearse pull-ci-medik8s-fence-agents-remediation-main-4.13-openshift-e2e

File: ci-operator/config/medik8s/fence-agents-remediation/medik8s-fence-agents-remediation-main__4.12.yaml [hunk 1]
```diff
@@ -28,25 +34,45 @@ resources:
...
+  commands: make test
+  container:
+    from: src
+  skip_if_only_changed: ^docs/|\.md$|^(?:.*/)?(?:\.gitignore|OWNERS|PROJECT|LICENSE)$
 - as: openshift-e2e
   skip_if_only_changed: ^docs/|\.md$|^(?:.*/)?(?:\.gitignore|OWNERS|PROJECT|LICENSE)$
   steps:
     cluster_profile: aws
+    dependencies:
+      OO_BUNDLE: my-bundle
```
Comment (clobrano) [inline ci-operator/config/medik8s/fence-agents-remediation/medik8s-fence-agents-remediation-main__4.12.yaml:47]: In other operators (e.g. NHC) the dependency is declared this way

```
dependencies:
- env: OO_BUNDLE
  name: my-bundle
```

are you sure this is correct as well?

**Comment by razo7** [comment]
/pj-rehearse pull-ci-medik8s-fence-agents-remediation-main-4.13-openshift-e2e

File: ci-operator/config/medik8s/fence-agents-remediation/medik8s-fence-agents-remediation-main__4.12.yaml (hunk 1 above)
Comment (razo7) [reply ci-operator/config/medik8s/fence-agents-remediation/medik8s-fence-agents-remediation-main__4.12.yaml:47]: Yes, I got this recommendation after communicating in [`forum-ocp-testplatform`](https://redhat-internal.slack.com/archives/CBN38N3MW/p1696253793949649).

Comment (razo7) [reply ci-operator/config/medik8s/fence-agents-remediation/medik8s-fence-agents-remediation-main__4.12.yaml:47]: It doesn't add the label from here :(

**Review by clobrano — APPROVED** [review]
/lgtm

**Pull request merged**